homework_4/eventsettler/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	stdhttp "net/http"
+	"time"
+)
+
+const eventUpdateEndpoint = "http://127.0.0.1:8080/event/update"
+const betsAPI = "http://127.0.0.1:9090/bets"
+const activeBetsParam = "?status=active"
+
+type Bet struct {
+	Id                   string  `json:"id"`
+	Status               string  `json:"status"`
+	SelectionId          string  `json:"selectionId"`
+	SelectionCoefficient float64 `json:"selectionCoefficient"`
+	Payment              float64 `json:"payment"`
+	Payout               float64 `json:"payout"`
+}
+
+func main() {
+	httpClient := stdhttp.Client{Timeout: time.Second * 10}
+
+	responseData, err := getActiveOdds(httpClient)
+	if err != nil {
+		log.Fatalln("error getting data: ", err)
+	}
+
+	events := getDistinctEvents(responseData)
+
+	eventUpdates := generateRandomOutcomes(events)
+
+	err = postEventUpdates(httpClient, eventUpdates)
+	if err != nil {
+		log.Fatalln("error sending event updates: ", err)
+	}
+
+}
+
+func postEventUpdates(httpClient stdhttp.Client, eventUpdates []EventUpdateDto) error {
+	for _, eventUpdate := range eventUpdates {
+		requestBody, err := json.Marshal(eventUpdate)
+		post, err := httpClient.Post(eventUpdateEndpoint, "text/plain", bytes.NewBuffer(requestBody))
```

Comment:
How come you put `text/plain` here when you are sending json objects?

Reply:
I got the idea from [here](https://github.com/superbet-group/code-cadets-2021/blob/main/lecture_4/01_event_api/internal/infrastructure/rabbitmq/eventUpdatePublisher.go). I thought the content-type should be the same as in that file because this post request gets published from that publisher. However, I agree it should be application/json, and I'm probably going to change both files if that's the right thing to do. The thing is, I'm unsure if channels can use application/json as content-type, but I dont't see a reason why not.

Reply:
I will wait for an explanation before I act on this.

Reply:
Makes sense that you would use it then 👍 . I think that in this case it doesn't matter too much. I was just curious why you decided to use it. I think it should work with `application/json`, but it works right now also 😁 . Also, you are sending bytes  through the queue which you then unmarshal, so there shouldn't be a problem with the content type.